src/metabase/models/setting.clj

```diff
@@ -1124,24 +1143,32 @@
            (map #(m/mapply user-facing-info % options)))
      (sort-by :name (vals @registered-settings)))))
 
+(defn can-read-setting?
+  "Returns true if the current user can read the setting, and false otherwise.
```

Comment:
nit: technically this function in isolation isn't tied to the current user, so maybe the wording can be tweaked here? just a suggestion
```suggestion
  "Returns true if a setting can be read according to the provided set of `allowed-visibilities`, and false otherwise.
```

Reply:
True, thanks